Python API docs build broken

The `py/Makefile` has problems post https://github.com/h2oai/wave/commit/1203878f7060eb12a09f99d1a0edf4e8b4274bce. `master` now fails me and @ashrith out of the box. 

One problem is that `h2o_wave_ml` is both a dir and a make task, so the task is never executed during a subsequent run if the first git clone/setup fails. So I tried patching it to run only during `make setup`...

```
diff --git a/py/Makefile b/py/Makefile
index 1fd4aa36..0e268209 100644
--- a/py/Makefile
+++ b/py/Makefile
@@ -11,16 +11,14 @@ build: purge
 setup: ## Install dependencies
        python3 -m venv venv
        ./venv/bin/python -m pip install --upgrade pip
+       git clone --depth 1 --branch $(WAVE_ML_VERSION) https://github.com/h2oai/wave-ml.git h2o_wave_ml
+       ./venv/bin/python -m pip install h2o_wave_ml/
        ./venv/bin/python -m pip install setuptools wheel httpx uvicorn starlette pdoc3 pytest flake8
        ./venv/bin/python -m pip install -r examples/requirements.txt
        ./venv/bin/python -m pip install --editable .

-h2o_wave_ml:
-       git clone --depth 1 --branch $(WAVE_ML_VERSION) https://github.com/h2oai/wave-ml.git h2o_wave_ml
-       cd h2o_wave_ml && ../venv/bin/python -m pip install .
-
 .PHONY: docs
-docs: h2o_wave_ml ## Build API docs
+docs: ## Build API docs
        ./venv/bin/pdoc --force  --template-dir docs/templates --output-dir build/docs h2o_wave
        ./venv/bin/pdoc --force  --template-dir docs/templates --output-dir build/docs h2o_wave_ml/h2o_wave_ml
        mkdir -p ../website/docs/api/h2o_wave_ml
```

...but then I run into this:

```
elp@ugh:~/git/h2oai/wave/py (master)
$ make clean setup
rm -rf build dist h2o_wave.egg-info h2o_wave_ml
rm -rf docs/build venv
python3 -m venv venv
./venv/bin/python -m pip install --upgrade pip
Collecting pip
  Using cached pip-21.0.1-py3-none-any.whl (1.5 MB)
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 20.2.3
    Uninstalling pip-20.2.3:
      Successfully uninstalled pip-20.2.3
Successfully installed pip-21.0.1
git clone --depth 1 --branch v0.4.0 https://github.com/h2oai/wave-ml.git h2o_wave_ml
Cloning into 'h2o_wave_ml'...
remote: Enumerating objects: 21, done.
remote: Counting objects: 100% (21/21), done.
remote: Compressing objects: 100% (16/16), done.
remote: Total 21 (delta 1), reused 12 (delta 0), pack-reused 0
Receiving objects: 100% (21/21), 103.86 KiB | 1.50 MiB/s, done.
Resolving deltas: 100% (1/1), done.
Note: switching to '9e9cc1e2a1a55a2d9671ff8665f0f56880822856'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

./venv/bin/python -m pip install h2o_wave_ml/
Processing ./h2o_wave_ml
Collecting h2osteam@ https://enterprise-steam.s3.amazonaws.com/release/1.8.2/python/h2osteam-1.8.2-py2.py3-none-any.whl
  Using cached https://enterprise-steam.s3.amazonaws.com/release/1.8.2/python/h2osteam-1.8.2-py2.py3-none-any.whl (56 kB)
Collecting mlops-client@ https://s3.amazonaws.com/artifacts.h2o.ai/releases/ai/h2o/mlops/rel-0.40.1/6/mlops_client-0.40.1%2Bea66172.rel0.40.1.12-py2.py3-none-any.whl
  Using cached https://s3.amazonaws.com/artifacts.h2o.ai/releases/ai/h2o/mlops/rel-0.40.1/6/mlops_client-0.40.1%2Bea66172.rel0.40.1.12-py2.py3-none-any.whl (390 kB)
Collecting datatable
  Using cached datatable-0.11.1.tar.gz (1.0 MB)
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... error
    ERROR: Command errored out with exit status 1:
     command: /home/elp/git/h2oai/wave/py/venv/bin/python /home/elp/git/h2oai/wave/py/venv/lib/python3.9/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmp_itqybfv
         cwd: /tmp/pip-install-f0pcovny/datatable_dfa3c4cb669140c4b1ebcc262052b42d
    Complete output (3 lines):
    Command `git rev-parse HEAD` failed with code 128: fatal: not a git repository (or any parent up to mount point /)
    Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).

    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/6a/4c/22f3be0cb1be3caaa737abd2d84dc98ebd9de45eb0cf15eb45f96c7690a7/datatable-0.11.1.tar.gz#sha256=3c3a7d3bee3c463a36fe77c5d4747278a741c23fc6c6e2dba110deec200502ba (from https://pypi.org/simple/datatable/) (requires-python:>=3.5). Command errored out with exit status 1: /home/elp/git/h2oai/wave/py/venv/bin/python /home/elp/git/h2oai/wave/py/venv/lib/python3.9/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmp_itqybfv Check the logs for full command output.
  Using cached datatable-0.11.0.tar.gz (1.0 MB)
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... error
    ERROR: Command errored out with exit status 1:
     command: /home/elp/git/h2oai/wave/py/venv/bin/python /home/elp/git/h2oai/wave/py/venv/lib/python3.9/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpt3ln_8dr
         cwd: /tmp/pip-install-f0pcovny/datatable_643092ec58e747a19bd2a5e7192dee7b
    Complete output (3 lines):
    Command `git rev-parse HEAD` failed with code 128: fatal: not a git repository (or any parent up to mount point /)
    Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).

    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/2e/6a/bba91ebcff851f7b650ddcf2783b997c7f9b4c2d46f7c90bd7e436396f99/datatable-0.11.0.tar.gz#sha256=19c602711e00f72e9ae296d8fa742d46da037c2d3a2d254bdf68f817a8da76bb (from https://pypi.org/simple/datatable/) (requires-python:>=3.5). Command errored out with exit status 1: /home/elp/git/h2oai/wave/py/venv/bin/python /home/elp/git/h2oai/wave/py/venv/lib/python3.9/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpt3ln_8dr Check the logs for full command output.
  Using cached datatable-0.10.1-cp39-cp39-linux_x86_64.whl
Collecting h2o==3.32.0.2
  Using cached h2o-3.32.0.2-py2.py3-none-any.whl
Collecting driverlessai
  Using cached driverlessai-1.9.2.1-py3-none-any.whl (1.6 MB)
Collecting requests
  Using cached requests-2.25.1-py2.py3-none-any.whl (61 kB)
Collecting future
  Using cached future-0.18.2-py3-none-any.whl

(snip)
```

The above command runs to completion despite errors, and running `make docs` fails:

```
elp@ugh:~/git/h2oai/wave/py (master)
$ make docs
./venv/bin/pdoc --force  --template-dir docs/templates --output-dir build/docs h2o_wave
/home/elp/git/h2oai/wave/py/venv/lib/python3.9/site-packages/pdoc/__init__.py:148: DeprecationWarning: Option `search_query` has been deprecated. Use `google_search_query` instead
  config = _get_config(**kwargs)
build/docs/h2o_wave/index.md
build/docs/h2o_wave/core.md
build/docs/h2o_wave/graphics.md
build/docs/h2o_wave/routing.md
build/docs/h2o_wave/server.md
build/docs/h2o_wave/test.md
build/docs/h2o_wave/types.md
build/docs/h2o_wave/ui.md
build/docs/h2o_wave/ui_ext.md
/home/elp/git/h2oai/wave/py/venv/lib/python3.9/site-packages/pdoc/cli.py:581: DeprecationWarning: Option `search_query` has been deprecated. Use `google_search_query` instead
  lunr_config = pdoc._get_config(**template_config).get('lunr_search')
./venv/bin/pdoc --force  --template-dir docs/templates --output-dir build/docs h2o_wave_ml/h2o_wave_ml
Traceback (most recent call last):
  File "/home/elp/git/h2oai/wave/py/venv/lib/python3.9/site-packages/pdoc/__init__.py", line 225, in import_module
    module = importlib.import_module(module_path)
  File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/home/elp/git/h2oai/wave/py/h2o_wave_ml/h2o_wave_ml/__init__.py", line 19, in <module>
    from .ml import Model, ModelType, ModelMetric, TaskType, build_model, get_model, save_model, load_model
  File "/home/elp/git/h2oai/wave/py/h2o_wave_ml/h2o_wave_ml/ml.py", line 25, in <module>
    import datatable as dt
  File "/home/elp/git/h2oai/wave/py/venv/lib/python3.9/site-packages/datatable/__init__.py", line 24, in <module>
    from .frame import Frame
  File "/home/elp/git/h2oai/wave/py/venv/lib/python3.9/site-packages/datatable/frame.py", line 23, in <module>
    from datatable.lib._datatable import Frame
  File "/home/elp/git/h2oai/wave/py/venv/lib/python3.9/site-packages/datatable/lib/__init__.py", line 31, in <module>
    from . import _datatable as core
ImportError: cannot import name '_datatable' from partially initialized module 'datatable.lib' (most likely due to a circular import) (/home/elp/git/h2oai/wave/py/venv/lib/python3.9/site-packages/datatable/lib/__init__.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/elp/git/h2oai/wave/py/./venv/bin/pdoc", line 33, in <module>
    sys.exit(load_entry_point('pdoc3==0.9.2', 'console_scripts', 'pdoc')())
  File "/home/elp/git/h2oai/wave/py/venv/lib/python3.9/site-packages/pdoc/cli.py", line 532, in main
    modules = [pdoc.Module(module, docfilter=docfilter,
  File "/home/elp/git/h2oai/wave/py/venv/lib/python3.9/site-packages/pdoc/cli.py", line 532, in <listcomp>
    modules = [pdoc.Module(module, docfilter=docfilter,
  File "/home/elp/git/h2oai/wave/py/venv/lib/python3.9/site-packages/pdoc/__init__.py", line 609, in __init__
    module = import_module(module)
  File "/home/elp/git/h2oai/wave/py/venv/lib/python3.9/site-packages/pdoc/__init__.py", line 227, in import_module
    raise ImportError('Error importing {!r}: {}: {}'
ImportError: Error importing 'h2o_wave_ml/h2o_wave_ml': ImportError: cannot import name '_datatable' from partially initialized module 'datatable.lib' (most likely due to a circular import) (/home/elp/git/h2oai/wave/py/venv/lib/python3.9/site-packages/datatable/lib/__init__.py)
make: *** [Makefile:23: docs] Error 1

```

--- a/py/Makefile
+++ b/py/Makefile
@@ -1,4 +1,4 @@
-WAVE_ML_VERSION := v0.4.0
+WAVE_ML_VERSION := v0.5.0
 
 all: build ## Build h2o_wave wheel
 
@@ -8,19 +8,21 @@ build: purge
 	sed -i -r -e "s/__version__.+/__version__ = '$(VERSION)'/" h2o_wave/__init__.py
 	./venv/bin/python3 setup.py bdist_wheel
 
-setup: ## Install dependencies
+.PHONY: prepare_wave_ml
+prepare_wave_ml:
+	rm -rf h2o_wave_ml
+	git clone --depth 1 --branch $(WAVE_ML_VERSION) https://github.com/h2oai/wave-ml.git h2o_wave_ml
+
+setup: prepare_wave_ml ## Install dependencies
 	python3 -m venv venv
 	./venv/bin/python -m pip install --upgrade pip
+	./venv/bin/python -m pip install h2o_wave_ml/
 	./venv/bin/python -m pip install setuptools wheel httpx uvicorn starlette pdoc3 pytest flake8
 	./venv/bin/python -m pip install -r examples/requirements.txt
 	./venv/bin/python -m pip install --editable .
 
-h2o_wave_ml:
-	git clone --depth 1 --branch $(WAVE_ML_VERSION) https://github.com/h2oai/wave-ml.git h2o_wave_ml
-	cd h2o_wave_ml && ../venv/bin/python -m pip install .
-
 .PHONY: docs
-docs: h2o_wave_ml ## Build API docs
+docs: ## Build API docs
 	./venv/bin/pdoc --force  --template-dir docs/templates --output-dir build/docs h2o_wave
 	./venv/bin/pdoc --force  --template-dir docs/templates --output-dir build/docs h2o_wave_ml/h2o_wave_ml
 	mkdir -p ../website/docs/api/h2o_wave_ml

